src/gov/nist/javax/sip/stack/SIPTransactionStack.java

```diff
@@ -1030,6 +1056,26 @@ public void removeDialog(String dialogId) {
         }
         dialogTable.remove(dialogId);
     }
+    
+    /**
+     * Adds a new dialog injection listener to this transaction.
+     * 
+     * @param newListener
+     *          Listener to add.
+     */
+    public void addDialogInjectionListener(SIPDialogInjectionListener newListener) {
+        dialogInjectionListeners.add(newListener);
```

Comment:
ok, then lets use a list potentially using Collections.synchronizedList to get the synchronized version. 
Also you may investigate to use the concurrent version CopyOnWriteArrayList, if the list is expected to be small (<10 elements) and modifications are low again...